plugins/flutter_aepcore/ios/Classes/FlutterAEPCoreDataBridge.m

```diff
@@ -89,4 +91,48 @@ + (NSDictionary *)sanitizeDictionaryToContainClass: (Class) type WithDictionary:
     return sanitizedDict;
 }
 
++ (AEPInitOptions *)initOptionsFromMap:(id)map {
+    if (![map isKindOfClass:[FlutterMethodCall class]]) {
+        return nil;
+    }
+
+    FlutterMethodCall *call = (FlutterMethodCall *)map;
+    NSDictionary *arguments = call.arguments;
+    if (![arguments isKindOfClass:[NSDictionary class]]) {
+        return nil;
+    }
+
+    NSDictionary *initOptionsMap = arguments[@"initOptions"];
+    if (![initOptionsMap isKindOfClass:[NSDictionary class]]) {
+        return nil;
+    }
+
+    NSString *appId = initOptionsMap[@"appId"];
+    NSNumber *lifecycleAutomaticTrackingEnabled = initOptionsMap[@"lifecycleAutomaticTrackingEnabled"];
+    NSDictionary *lifecycleAdditionalContextData = initOptionsMap[@"lifecycleAdditionalContextData"];
+    NSString *appGroup = initOptionsMap[@"appGroup"];
+
+    AEPInitOptions *initOptions;
+    if (appId != nil && ![appId isKindOfClass:[NSNull class]]) {
+        initOptions = [[AEPInitOptions alloc] initWithAppId:appId];
+    } else {
+        initOptions = [[AEPInitOptions alloc] init];
+    }
+    if (lifecycleAutomaticTrackingEnabled != nil && [lifecycleAutomaticTrackingEnabled isKindOfClass:[NSNumber class]] ) {
+        initOptions.lifecycleAutomaticTrackingEnabled = [lifecycleAutomaticTrackingEnabled boolValue];
+    }
+    if (lifecycleAdditionalContextData != nil && [lifecycleAdditionalContextData isKindOfClass:[NSDictionary class]]) {
+        initOptions.lifecycleAdditionalContextData = lifecycleAdditionalContextData;
+    } else {
+        NSLog(@"lifecycleAdditionalContextData is nil or not a dictionary");
+    }
+    if (appGroup != nil && [appGroup isKindOfClass:[NSString class]]) {
+        initOptions.appGroup = appGroup;
+    } else {
+        NSLog(@"appGroup is nil or not a string");
```

Comment:
I'd recommend removing this log entry as it's probably more common to not set an app group.